modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -245,9 +245,10 @@
   [[_ value {base-type :base_type}]]
   (if (and (isa? base-type :type/MongoBSONID)
            (some? value))
-    ;; Passing a nil to the ObjectId constructor throws an exception
+    ;; Passing nil or "" to the ObjectId constructor throws an exception
     ;; "invalid hexadecimal representation of an ObjectId: []" so, just treat it as nil
-    (ObjectId. (str value))
+    (when (not= value "")
```

Comment:
This kinda confuses me as to what types of values flow through. We check for empty string and then call `str` on the value. I'm surprised that seemingly more than just strings are flowing through this. What values come through this path?

Reply:
I suppose constants provided by the user and values provided by the query processor. From the latter group I only know about `""` used when checking emptiness.

Reply:
I'm still not following. We're checking `value` against the empty string and then calling `str` on it. What types flow through here (or do we know?). Obviously strings can come through if we check against the empty string. Do integers come through? Maps, vectors, etc.